### DISCUSSION TITLE
Is there any example of streaming videos with Supabase?

### DISCUSSION THREAD

**Discussion by RatulSaha** [supabase/supabase#5566] [Questions] [answered]
From https://supabase.com/blog/2021/07/27/storage-beta, it suggests adding the header. Is there anything else to do, maybe for conversion/codec etc?

Would really appreciate if there is a hello world on this with Supabase (coming from a AWS MediaConvert kind of experience).

**Comment by fenos** [comment, marked as answer]
If you are referring to live stream, we don't currently support that.

If you are to simply play a video you can upload an mp4 to storage, you will be able to play the video just fine in the HTML `<video>` tag.

Alternatively, you can upload an HLS video to storage and use hls.js.
We currently don't support automatic video encoding or transmuxing but is in the roadmap

**Reply by itsmikesharescode** [reply]
damn supabase is so awsome! please dont be like platnetscale

**Reply by Akash187** [reply]
Hey @itsmikesharescode you can try Cloudflare Stream for a ready-made solution. If you want a cheaper alternative then you can go with Cloudflare R2 which I am actually using. R2 requires a bit of work to set up for video streaming. Let me know if you need any further help.

Supabase is an excellent serverless Postgresql database. Kudos to the Supabase team.

**Reply by dinhkhanh** [reply]
@fenos is there any updates related to this part: "support automatic video encoding or transmuxing"

**Reply by rootacc3ss** [reply]
@fenos bump on @dinhkhanh's question

**Comment by Akash187** [comment]
Just for an update As I have tested video streaming is just working fine by default.
```
<video
	controls
	controlsList="nodownload"
	autoPlay
	style={{ width: '100%', maxWidth: '1080px', display: 'block' }}>
	<source
		src="link to the video on supabase"
		type="video/mp4"
	/>
</video>
```

**Comment by drummerjolev** [comment]
I'd like to play an uploaded video, but only for an authenticated user. How can I do that? I understand a signed URL will make the video accessible publicly, but I want the viewer to be authenticated.

**Reply by axelrindle** [reply]
You can achieve this by creating a policy for the bucket the videos are uploaded to.

![image](https://github.com/supabase/supabase/assets/7105632/c1460c03-cef7-4c66-96d5-69ea254db837)

**Reply by drummerjolev** [reply]
Thanks @axelrindle – that doesn't work though, because I can't send an Authentication header in `<video src="{PROTECTED_URL}" ...`. Any other ideas to deal with this use case?

**Reply by axelrindle** [reply]
You don't need the `Authorization` header for the video tag, because you use the signed URL as the `src`. Generating the signed URL however will be protected by the policy. So unauthorized user won't be able to generate a signed URL at all.

**Reply by drummerjolev** [reply]
But the content will be viewable through this URL by anyone who has it – not only by e.g. users behind auth. Is there no way around the signed URL?

I want to support the following flow:
* Login
* Upload a video
* View your own videos (ideally, without a signed URL that can shared/viewed outside the authed platform)

**Reply by axelrindle** [reply]
I'm afraid there is no other way I know of to accomplish this.

> But the content will be viewable through this URL by anyone who has it – not only by e.g. users behind auth.

The signed URL won't be valid forever. The token is only valid for a given amount of time. You can set this lifetime to be 5 minutes for example.

**Reply by Akash187** [reply]
I will suggest Cloudflare Stream if you need a ready-made solution. else you can use Cloudflare R2.

**Reply by drummerjolev** [reply]
Thanks @Akash187, I'll give those a try!
